x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MlTrainedModelConfig } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+import { MlInferencePipeline } from '../types/pipelines';
+
+// Getting an error importing this from @kbn/ml-plugin/common/constants/data_frame_analytics'
+// So defining it locally for now with a test to make sure it matches.
+export const BUILT_IN_MODEL_TAG = 'prepackaged';
+
+export interface MlInferencePipelineParams {
+  description?: string;
+  destinationField: string;
+  model: MlTrainedModelConfig;
+  pipelineName: string;
+  sourceField: string;
+}
+
+/**
+ * Generates the pipeline body for a machine learning inference pipeline
+ * @param pipelineConfiguration machine learning inference pipeline configuration parameters
+ * @returns pipeline body
+ */
+export const generateMlInferencePipelineBody = ({
+  description,
+  destinationField,
+  model,
+  pipelineName,
+  sourceField,
+}: MlInferencePipelineParams): MlInferencePipeline => {
+  // if model returned no input field, insert a placeholder
+  const modelInputField =
+    model.input?.field_names?.length > 0 ? model.input.field_names[0] : 'MODEL_INPUT_FIELD';
+  return {
+    description: description ?? '',
+    processors: [
+      {
+        remove: {
+          field: `ml.inference.${destinationField}`,
+          ignore_missing: true,
+        },
+      },
+      {
+        inference: {
+          field_map: {
+            [sourceField]: modelInputField,
```

Comment:
I think we'll probably need to refactor this to take the `field_map` as an input eventually to support selecting N source fields in the UI.